systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -11,19 +11,57 @@
 %
 
 if model.use_new_kinsol
-  checkDirty(model);
-  if nargin<5, qd=[]; end
-  if nargin<4, use_mex = true; end
+  checkDirty(model);  
+  if nargin<6
+    options = struct();
+    if nargin > 2
+      options.compute_gradients = b_compute_second_derivatives;
+    end; % approximately the same use case
+    if nargin > 3
+      options.use_mex = use_mex;
+    end;
+  else
+    % if options struct is passed in, make sure that boolean arguments are
+    % empty to avoid conflicts
+    if nargin > 2
+      valuecheck(b_compute_second_derivatives, []);
+    end;
+    if nargin > 3
+      valuecheck(use_mex, []); 
+    end
+  end
+  if nargin > 4
+    v = qd; % TODO: v is what should be passed to doKinematics
+  else
+    v = [];
+  end
+  if ~isfield(options, 'use_mex')
```

Comment:
i guess it would be good to make sure that we're not taking a performance hit here.  but again, i much prefer the cleaner input arguments.